### ISSUE TITLE
Archiving this codebase on Arweave for permanence

### ISSUE THREAD

**Issue by t8** [solana-labs/solana#34769]
#### Problem

Solana's core codebases are currently only stored in web2 environments, which aren't permanent nor immutable.

#### Proposed Solution

Would like to propose automatically archiving this repo to Protocol.land, a decentralized source control protocol on Arweave.

A large part of the Solana community already leverages Arweave for data storage, so thinking it might make sense to do the same for the underlying codebase 🙂

If so, let me know and I’d be happy to open a PR! Super simple to do with a Github action: https://docs.protocol.land/working-with-repositories/import-a-repository-from-github

**Comment by Himnasmidur** [comment]
This makes a lot of sense. I support it! 

Solana's speed with Arweave's stability would be a formidable combination.

**Comment by 0legRadchenko** [comment]
yep, good idea to safeguard Solana's codebase with Protocol.land's immutable storage :)

**Comment by whyyyseee** [comment]
I like this idea, makes sense for Solana's codebase to be archived on protocol.land instead of a traditional web2 solution.

**Comment by angrymouse** [comment]
Hmm, what's wrong with just taking code snapshot every few months and uploading .tar.gz to arweave? @t8

**Comment by angrymouse** [comment]
Running upload script on every push means flooding uploading a lot of txs to arweave, which may be not very cost effective option. Proposed CI script in docs isn't most secure either, as it runs *newest* version of @protocol.land/sync passing github auth token and arweave wallet to it.
![image](https://github.com/solana-labs/solana/assets/40735471/587963d8-1d28-4411-b3ae-238b09b5df44)
Meaning if someone would compromise access to @protocol.land/sync on NPM, it would allow them to get github access token and arweave wallet from this repo's secrets.

**Comment by angrymouse** [comment]
Though from security perspective, I believe just switching to forced version would be fine (for example to @protcol.land/sync@0.1.4). From practical perspective tho, I think solana validators already hold full copies of nodes' code for it to be earthquake-proof, though uploading to arweave might add even more resilience.

**Comment by kevinheavey** [comment]
Don't worry guys I have a snapshot on my laptop, no need for arweave
